src/parser.ts

```diff
@@ -53,9 +59,15 @@ function resetExtra(): void {
  */
 function getASTFromProject(code: string, options: ParserOptions) {
   return util.firstDefined(
-    calculateProjectParserOptions(code, options.filePath, extra),
+    calculateProjectParserOptions(
+      code,
+      options.filePath || (options.jsx ? 'estree.ts' : 'estree.tsx'),
+      extra
+    ),
     (currentProgram: ts.Program) => {
-      const ast = currentProgram.getSourceFile(options.filePath);
+      const ast = currentProgram.getSourceFile(
+        options.filePath || (options.jsx ? 'estree.ts' : 'estree.tsx')
```

Comment:
This filename logic is now separately defined 3 times within this file, can we promote it to be defined once at a higher level?

Other than that the PR LGTM

Reply:
Updated.